pkg/virt-controller/watch/vm_test.go

```diff
@@ -5640,39 +5879,26 @@ var _ = Describe("VirtualMachine", func() {
 				restartRequired = false
 			})
 
-			AfterEach(func() {
-				k8sClient.Fake.ClearActions()
-			})
-
 			It("should appear when changing a non-live-updatable field", func() {
 				testutils.UpdateFakeKubeVirtClusterConfig(kvInformer, kv)
 
-				By("Creating a VM with hostname a")
+				By("Creating a VMI with hostname 'a'")
 				vm.Spec.Template.Spec.Hostname = "a"
-				addVirtualMachine(vm)
-
-				By("Executing the controller expecting a VMI to get created and no RestartRequired condition")
 				vmi = controller.setupVMIFromVM(vm)
-				expectVMICreation()
-				expectVMStatusUpdate()
-				expectControllerRevisionList()
-				expectControllerRevisionCreation()
-				sanityExecute(vm)
-				syncCaches()
-				Consistently(restartRequired, 1*time.Second).Should(BeFalse(), "restart required")
-				markAsReady(vmi)
-				vmiFeeder.Add(vmi)
+				vmiSource.Add(vmi)
```

Comment:
Some tests in vmi_test.go use the suggestion below, and seem to have less/no timing issue. Maybe something to explore?
```suggestion
				Expect(vmiInformer.GetIndexer().Add(vmi)).To(Succeed())
```